doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -266,6 +360,74 @@ More on sorts can be found in Section :ref:`sorts`.
 Binders
 -------
 
+.. insertgram binder_constr name_list
+
+.. productionlist:: coq
+   binder_constr : fun `open_binders` => `term`
+                 : forall `open_binders` , `term`
+                 : let `name` `binder_list_opt` `type_cstr` := `term` in `term`
+                 : let `single_fix` in `term`
+                 : let `name_alt` `return_type_opt` := `term` in `term`
+                 : let ' `pattern` := `term` in `term`
+                 : let ' `pattern` := `term` return `term100` in `term`
+                 : let ' `pattern` in `pattern` := `term` return `term100` in `term`
+                 : let : `pattern` := `term` in `term`      (ssr plugin)
+                 : let : `pattern` := `term` `ssr_rtype` in `term`      (ssr plugin)
+                 : let : `pattern` in `pattern` := `term` `ssr_rtype` in `term`      (ssr plugin)
+                 : if `term` `return_type_opt` then `term` else `term`
+                 : `single_fix`
+                 : `single_fix` with `fix_body_list` for `ident`
+                 : if `term` is `ssr_dthen` `ssr_else`      (ssr plugin)
+                 : if `term` isn't `ssr_dthen` `ssr_else`      (ssr plugin)
+   type_cstr : `term_opt`
+             : : `term`
+             : `empty`
+   term_opt : : `term`
+            : `empty`
+   name : _
+        : `ident`
+   name_alt : ( `name_list_comma_opt` )
+            : ()
+   name_list_comma_opt : `name_list_comma`
+                       : `empty`
+   name_list_comma : `name_list_comma` , `name`
+                   : `name`
+   open_binders : `name` `name_list_opt` : `term`
+                : `name` `name_list_opt` `binder_list_opt`
+                : `name` .. `name`
+                : `closed_binder` `binder_list_opt`
+   name_list_opt : `name_list_opt` `name`
+                 : `empty`
+   binder_list_opt : `binder_list_opt` `binder`
+                   : `empty`
+   binder : `name`
+          : `closed_binder`
+   closed_binder : ( `name` `name_list` : `term` )
+                 : ( `name` : `term` )
+                 : ( `name` := `term` )
+                 : ( `name` : `term` := `term` )
+                 : { `name` }
+                 : { `name` `name_list` : `term` }
+                 : { `name` : `term` }
+                 : { `name` `name_list` }
+                 : `( `typeclass_constraint_list_comma` )
+                 : `{ `typeclass_constraint_list_comma` }
+                 : ' `pattern0`
+                 : `of_alt` `term10`      (ssr plugin)
+                 : ( _ : `term` | `term` )
+   return_type_opt : `as_name_opt` return `term100`
+                   : `empty`
+   as_name_opt : as `name`
+               : `empty`
+   single_fix : `fix_kw` `fix_body`
+   fix_body_list : `fix_body_list` with `fix_body`
+                 : `fix_body`
+   fix_kw : fix
+          : cofix
```

Comment:
Given how `fix_kw` is only used once, I'd like to inline its definition so that `single_fix` is either ``` fix `fix_body` ``` or ``` cofix `fix_body` ```. It will make reading easier.

Reply:
productionlist supports standard BNF grammars.  The only way to represent alternatives in such grammars is to create a separate nonterminal with the alternatives.  Using `[`, `|` and `]` both as metacharacters and to represent themselves would be ambiguous.  Eventually using `prodn::` will allow inlining.

Reply:
I was suggesting the following:

```
single_fix : fix `fix_body`
             cofix `fix_body`
```

Repetitive, but still better.

Reply:
got it